sonaranalyzer-dotnet/src/SonarAnalyzer.Common/Rules/DoNotHardcodeCredentialsBase.cs

```diff
@@ -0,0 +1,127 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2018 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using SonarAnalyzer.Common;
+using SonarAnalyzer.Helpers;
+
+namespace SonarAnalyzer.Rules
+{
+    public abstract class DoNotHardcodeCredentialsBase : ParameterLoadingDiagnosticAnalyzer
+    {
+        protected const string DiagnosticId = "S2068";
+        protected const string MessageFormat = "Remove hard-coded password(s): '{0}'.";
+
+        private const string DefaultCredentialWords = "password, passwd, pwd";
+
+        private string credentialWords;
+        private IEnumerable<string> splitCredentialWords;
+        private Regex passwordValuePattern;
+        
+        [RuleParameter("credentialWords", PropertyType.String,
+            "Comma separated list of words identifying potential credentials", DefaultCredentialWords)]
+        public string CredentialWords
+        {
+            get => this.credentialWords;
```

Comment:
this isn't covered by tests - is it used?

Reply:
It is not used indeed, but it is silly to create a property with just a setter (SQ will also complain)... Also CredentialWords is required to be a property, so we don't have much choice.

Reply:
I'd rather have SQ complaining (and mark as Won't fix) than having useless code